test/integration/scheduler/queue_test.go

```diff
@@ -1290,6 +1298,653 @@ func TestCoreResourceEnqueue(t *testing.T) {
+					Obj()},
+			initialPVCs: []*v1.PersistentVolumeClaim{
+				st.MakePersistentVolumeClaim().
+					Name("pvc1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteMany}).
+					Resources(v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("2Mi")}}).
+					StorageClassName(ptr.To("sc1")).
+					Obj(),
+			},
+			initialStorageClasses: []*storagev1.StorageClass{
+				st.MakeStorageClass().
+					Name("sc1").
+					Label("key", "value").
+					VolumeBindingMode(storagev1.VolumeBindingWaitForFirstConsumer).
+					Provisioner("p").
+					AllowedTopologies([]v1.TopologySelectorTerm{
+						{
+							MatchLabelExpressions: []v1.TopologySelectorLabelRequirement{
+								{Key: v1.LabelTopologyZone, Values: []string{"us-west1-a"}},
+							},
+						},
+					}).Obj(),
+			},
+			pods: []*v1.Pod{
+				st.MakePod().Name("pod1").Container("image").PVC("pvc1").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
+				sc1 := st.MakeStorageClass().
+					Name("sc1").
+					Label("key", "updated").
+					VolumeBindingMode(storagev1.VolumeBindingWaitForFirstConsumer).
+					Provisioner("p").
+					AllowedTopologies([]v1.TopologySelectorTerm{
+						{
+							MatchLabelExpressions: []v1.TopologySelectorLabelRequirement{
+								{Key: v1.LabelTopologyZone, Values: []string{"us-west1-a"}},
+							},
+						},
+					}).
+					Obj()
+				if _, err := testCtx.ClientSet.StorageV1().StorageClasses().Update(testCtx.Ctx, sc1, metav1.UpdateOptions{}); err != nil {
+					return nil, fmt.Errorf("failed to update storageclass: %w", err)
+				}
+				return map[framework.ClusterEvent]uint64{framework.StorageClassUpdate: 1}, nil
+			},
+			wantRequeuedPods:          sets.Set[string]{},
+			enableSchedulingQueueHint: []bool{true},
+		},
+		{
+			name:         "Pod rejected with node by the VolumeBinding plugin is requeued when the CSINode is created",
+			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node").Label("node", "fake-node").Obj()},
+			initialPVs: []*v1.PersistentVolume{
+				st.MakePersistentVolume().
+					Name("pv1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteMany}).
+					Capacity(v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}).
+					HostPathVolumeSource(&v1.HostPathVolumeSource{Path: "/tmp", Type: ptr.To(v1.HostPathDirectoryOrCreate)}).
+					Obj()},
+			initialPVCs: []*v1.PersistentVolumeClaim{
+				st.MakePersistentVolumeClaim().
+					Name("pvc1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteMany}).
+					Resources(v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("2Mi")}}).
+					Obj(),
+			},
+			pods: []*v1.Pod{
+				st.MakePod().Name("pod1").Container("image").PVC("pvc1").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
+				csinode1 := st.MakeCSINode().Name("csinode").Obj()
```

Comment:
Shouldn't this csinode be referenced anywhere above?

Reply:
I made the CSINode and Node have the same name.